File: est datos/cpp/busqueda.cpp
#include <iostream>
using namespace std;

int buscar(int arr[], int tam, int dato) {
for (int i = 0 ; i<tam ; i++){
    if (arr[i] == dato)
    {
        return i;
    }
}
return -1;
}


int main(int argc, char const *argv[])
{
    int tam, dato, j;
    cout << "De que tamaño desea su arreglo?" << endl;
    cin >> tam;

    cout << "Se lleno el arreglo con numeros random desde 1 hasta 100" << endl;

    int arr[tam];

    for (int i = 0; i< tam ; i++){
        arr[i] = rand() % 100 + 1;
    }


    do
    {
        
    cout << "Ingrese el numero que desea buscar dentro del arreglo: " << endl;
    cin >> dato;

    int position = buscar(arr, tam, dato);

         if (position != -1){
        cout << "El numero " << dato << " se encuentra en la posicion " << (position+1) << " del arreglo." << endl;
    } else{
        cout << "El numero " << dato << " no se encuentra en el arreglo." << endl;
    }

    
    cout << "Desea buscar otro numero? (1 para si, 0 para no)" << endl;
    cin >> j;

    } while (j == 1);
    
    cout << "Gracias por participar!" << endl;

    return 0;
}

File: est datos/Js/Busqueda.js
function busqueda(arr, dato,tam){
    for (let i = 0; i<tam ; i++){
        if (arr[i] == dato){
            return i;
        }
    }
    return -1;

}

let tam = parseInt(prompt("Ingrese el tamaño del arreglo"));

let arr = new Array(tam);

for (let i = 0 ; i<tam ; i++){
    arr[i] = Math.floor(Math.random()*100)+1;
}

alert("El arreglo es: " + arr);

alert("Se lleno el arreglo de numeros random entre 1 y 100");

let j;

do {
    let dato = prompt("Ingrese el numero a buscar en el arreglo: ");
    let resultado = busqueda(arr,dato,tam);

    if (resultado != -1){
        alert("El numero " + dato + " se encuentra en la posicion " + (resultado+1))
    } else {
        alert("El numero " + dato + " no se encuentra en el arreglo")
    }

    j = parseInt(prompt("Desea buscar otro numero? 1.Si 2.No"));

}while (j==1)

    alert("Gracias por usar el programa");